### ISSUE TITLE
Large numbers to text  (1200 -> 1.2K)

### ISSUE THREAD

**Issue by RohitEdathil** [kentcb/humanizer#20]
In many sites and apps, they show large numbers in K, M, B etc. right ? I think It would be a nice feature to have. I actually had to do something similar manually recently for [a project of mine ](https://github.com/RohitEdathil/project_card). Can I port it into Dart and maybe open a PR to this repo? Here is [my implementation](https://github.com/RohitEdathil/project_card/blob/master/src/formatters.py) in python BTW.

**Comment by kentcb** [comment]
Hi @RohitEdathil, thanks for the suggestion (and for offering to help!).

Hmm, it's interesting because I feel like the library _almost_ has this already. The Units of Measurement feature basically does all this, except it currently requires you to choose an appropriate unit. In your scenario, what does the value represent? Does it fall nearly into one of the units of measure? If not, I'm wondering whether we need a more general `Quantity` UoM.

**Comment by RohitEdathil** [comment]
I was thinking of a more general case, like the number of likes, comments etc.

**Comment by RohitEdathil** [comment]
I've decided to create a new package with this feature and some more.  Thanks for your time 😀

**Comment by kentcb** [comment]
NOTE: this is now available in V2.0.0:

```dart
final quantity = 1.million() + 100.thousand();

// '1.1M'. Can also use QuantityFormat to get more control over formatting.
final formatted = quantity.toString();
```
